test-wasm.html

```diff
@@ -0,0 +1,102 @@
+<script src="wasm/harfbuzzjs.js"></script>
+<script>
+var decoder = new TextDecoder("utf8");
+function bufferToString(buffer) {
+  return decoder.decode(buffer.slice(0, buffer.indexOf(0)));
+}
+
+function hb_tag(tag) {
+  return [...tag].reduce((x, y) => (x << 8) + y.codePointAt(0), 0);
+}
+
+Module().then(module => {
+  window.module = module;
+
+  fetch('test/subset/data/fonts/Roboto-Regular.ttf').then(res => res.arrayBuffer()).then(font => {
+    var buffer = module._malloc(font.byteLength);
+    module.HEAPU8.set(new Uint8Array(font), buffer);
+
+    var blob = module._hb_blob_create(buffer, font.byteLength, 0/*FIXME*/, 0, 0);
+    // module._hb_blob_get_length(blob), module._hb_face_count(blob)
+    var face = module._hb_face_create(blob, 0);
+    var font = module._hb_font_create(face);
+
+    var buffer = module._hb_buffer_create();
+    [...'Your text to shape!'].forEach((x, i) => {
+      module._hb_buffer_add(buffer, x.codePointAt(0), i);
+    });
+    module._hb_buffer_set_content_type(buffer, 1/*HB_BUFFER_CONTENT_TYPE_GLYPHS*/);
```

Comment:
This line is not needed.

Reply:
Fails with `uncaught exception: abort("Assertion failed: content_type == HB_BUFFER_CONTENT_TYPE_UNICODE || (!len && content_type == HB_BUFFER_CONTENT_TYPE_INVALID), at: ../src/hb-buffer.cc,642,guess_segment_properties"). Build with -s ASSERTIONS=1 for more info.` if I don't.

Reply:
Actually.  Humm.  You setting it to CONTENT_TYPE_GLYPHS?!!!  Your comment is wrong.  1 is CONTENT_TYPE_UNICODE.  The question is, why did your buffer have CONTENT_TYPE_GLYPHS in it otherwise?  Did you not reset the buffer after calling shape on it once?

Reply:
Hmm, I don't call it more than once, let me use hb_buffer_add_utf8 to resolve this :)

Reply:
Oh you're right.  It's the `add_utfX` that set the content type to Unicode.  Nvm.